### PR TITLE
First approach to a CI

### PR DESCRIPTION
**Pull request by aleixpol**
Compiles the project on debian stretch and fedora

See results: https://travis-ci.org/aleixpol/PackageKit (will have to enable it on hugsie after merging)

### CODE REVIEW

File: tests/ci/Dockerfile-fedora [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM fedora
+
+RUN dnf -y update
+RUN dnf -y install dnf-plugins-core
+RUN dnf -y builddep PackageKit
+RUN dnf -y install libdnf-devel redhat-rpm-config autoconf-archive
+
```
Comment (Conan-Kudo) [inline tests/ci/Dockerfile-fedora]: I'd merge lines 4 and 6 together, and also add `autoconf automake libtool` to that line.

File: tests/ci/Dockerfile-fedora [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM fedora
```
Comment (Conan-Kudo) [inline tests/ci/Dockerfile-fedora]: `FROM fedora:latest`

File: tests/ci/Dockerfile-fedora (hunk 1 above)
Comment (Conan-Kudo) [inline tests/ci/Dockerfile-fedora]: Also, you can make them all quiet by adding `-q` to `-y`.

File: tests/ci/Dockerfile-fedora (hunk 2 above)
Comment (aleixpol) [inline tests/ci/Dockerfile-fedora]: I wonder if we want to specify the version like on debian...

Comment (Conan-Kudo) [inline tests/ci/Dockerfile-fedora]: Id you'd want to do that, maybe set `25`.

**Comment by hughsie** [comment]
Looks great! How do we set up the "hook" to make travis re-run the tests on new checkins?

**Comment by ximion** [comment]
@hughsie You need to select the Travis module (and register at Travis) in Github's repo configuration. The rest is done automatically by GH and Travis.

**Comment by aleixpol** [comment]
Ping

**Comment by hughsie** [comment]
Sorry about this, currently at a conference. I like the idea, we just have to make it work! :)

**Comment by ximion** [comment]
This is also just building stuff at time, and not yet running tests - running the unit-tests would be pretty awesome, but in case of PK it's also an incredibly difficult task (esp. for the performance tests).

**Comment by hughsie** [comment]
Okay, merge done. I guess setting up travis is next, right?

**Pull request merged**